components/compositing/compositor.rs

```diff
@@ -772,7 +809,82 @@ impl IOCompositor {
         input_event_id: InputEventId,
         result: InputEventResult,
     ) {
-        self.painter_mut(webview_id.into())
-            .notify_input_event_handled(webview_id, input_event_id, result);
+        if let Some(mut painter) = self.maybe_painter_mut(webview_id.into()) {
+            painter.notify_input_event_handled(webview_id, input_event_id, result);
+        }
+    }
+
+    /// Generate an image key from the appropriate [`Painter`] or, if it is unknown, generate
+    /// a dummy image key. The unknown case needs to be handled because requests for keys
+    /// could theoretically come after a [`Painter`] has been released. A dummy key is okay
+    /// in this case because we will never render again in that case.
+    fn handle_generate_image_key(
+        &self,
+        webview_id: WebViewId,
+        result_sender: GenericSender<ImageKey>,
+    ) {
+        let painter_id = webview_id.into();
+        let image_key = self.maybe_painter(painter_id).map_or_else(
+            || ImageKey::new(painter_id.into(), 0),
+            |painter| painter.webrender_api.generate_image_key(),
+        );
+        let _ = result_sender.send(image_key);
+    }
+
+    /// Generate image keys from the appropriate [`Painter`] or, if it is unknown, generate
+    /// dummy image keys. The unknown case needs to be handled because requests for keys
+    /// could theoretically come after a [`Painter`] has been released. A dummy key is okay
+    /// in this case because we will never render again in that case.
+    fn handle_generate_image_keys_for_pipeline(
+        &self,
+        webview_id: WebViewId,
+        pipeline_id: PipelineId,
+    ) {
+        let painter_id = webview_id.into();
+        let painter = self.maybe_painter(painter_id);
+        let image_keys = (0..pref!(image_key_batch_size))
```

Comment:
Is there any reason we need to return dummy image keys rather than just an empty vector?